client/config.c

```diff
@@ -192,6 +242,32 @@ TDNFReadConfig(
         }
     }
 
+    dwError = TDNFAllocateMemory(UserAgentHeaderMaxLength, sizeof(char), (void**)&pszUserAgentHeader);
+    BAIL_ON_TDNF_ERROR(dwError);
+
+	strcpy(pszUserAgentHeader, "User-Agent:tdnf/");
```

Comment:
`"User-Agent:tdnf/"` storing these keywords in macros would help

Reply:
If we use `TDNFAllocateStringPrintf()` it will be a format string.